src/libraries/System.Net.Http/tests/UnitTests/Headers/HeaderEncodingTest.cs

```diff
@@ -24,12 +24,12 @@ public void RoundTripsUtf8(string input)
             byte[] encoded = Encoding.UTF8.GetBytes(input);
 
             Assert.True(HeaderDescriptor.TryGet("custom-header", out HeaderDescriptor descriptor));
-            Assert.Null(descriptor.KnownHeader);
+            Assert.IsType<string>(descriptor.Descriptor);
```

Comment:
Why is using Descriptor here better than using KnownHeader, as previously? Is it just that KnownHeader got removed?

Reply:
(and similarly elsewhere)

Reply:
Right, I went through the places that used `KnownHeader` and changed them to do `ReferenceEquals` checks based on `descriptor.Descriptor`, since a `KnownHeader` property would incur a type check. After that, there were no more uses of it in the product code so I just removed it.

Reply:
`Descriptor` has been removed. These now check `HasValue` and `IsKnownHeader`